rust/noosphere/src/wasm/sphere.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::Result;
+use cid::Cid;
+use std::sync::Arc;
+
+use crate::{
+    platform::{PlatformKeyMaterial, PlatformStorage},
+    sphere::SphereContext as SphereContextImpl,
+    wasm::SphereFs,
+};
+use tokio::sync::Mutex;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+/// A `SphereContext` is a view into all of a sphere's data, that also
+/// encapsulates handles to local storage and a user's authority relative to the
+/// sphere. If a user is appropriately authorized, they may use a
+/// `SphereContext` to modify a sphere. Otherwise, they may only read a sphere's
+/// publicly visible content.
+pub struct SphereContext {
+    #[wasm_bindgen(skip)]
+    pub inner: Arc<Mutex<SphereContextImpl<PlatformKeyMaterial, PlatformStorage>>>,
```

Comment:
How do mutexes work via wasm bindgen? Atomics and SABs? Curious about the runtime implementation of this!

Reply:
Yeah, `wasm-bindgen` brings along a single-threaded async runtime. The fact that we can drop-in some alternative runtime is the silver lining of super-low-level Rust async, and so we take advantage of it here. It generally means that anything we might have otherwise been able to parrallelize using futures becomes at best concurrent.

Reply:
That sounds way easier than COOP/COEP configuration, great